chainerx_cc/chainerx/python/routines.cc

```diff
@@ -409,13 +409,86 @@ void InitChainerxManipulation(pybind11::module& m) {
           py::arg("arrays"),
           py::arg("axis") = 0);
     m.def("split",
-          [](const ArrayBodyPtr& ary, int64_t sections, int8_t axis) { return MoveArrayBodies(Split(Array{ary}, sections, axis)); },
-          py::arg("ary"),
-          py::arg("indices_or_sections"),
-          py::arg("axis") = 0);
-    m.def("split",
-          [](const ArrayBodyPtr& ary, std::vector<int64_t> indices, int8_t axis) {
-              return MoveArrayBodies(Split(Array{ary}, indices, axis));
+          [](const ArrayBodyPtr& ary, py::handle indices_or_sections, int8_t axis) {
+              auto split_sections = [](const ArrayBodyPtr& ary, int64_t sections, int8_t axis) {
+                  return MoveArrayBodies(Split(Array{ary}, sections, axis));
+              };
+              auto split_indices = [](const ArrayBodyPtr& ary, const std::vector<int64_t>& indices, int8_t axis) {
+                  return MoveArrayBodies(Split(Array{ary}, indices, axis));
+              };
+
+              // Converts an python float to sections (int64_t).
+              // Raises ValueError if the value has non-zero fraction.
+              auto pyfloat_to_sections_or_value_error = [](py::handle num) {
+                  CHAINERX_ASSERT(py::isinstance<py::float_>(num));
+                  double num_fp = py::cast<double>(num);
+                  auto num_int = static_cast<int64_t>(num_fp);
+                  if (static_cast<double>(num_int) != num_fp) {
+                      throw py::value_error{"Sections must be an integer."};
+                  }
+                  return num_int;
+              };
```

Comment:
How about defining these as free functions since they're only used conditionally?

Reply:
I assume their cost for definition is zero because they have no capture.
I think it's better to define here because they're currently used only in this function.

Reply:
Okay, let's leave it as is.